src/App.js

```diff
@@ -33,32 +33,69 @@ const NoLogin = () => {
   )
 }
 
+const Error = (error) => {
+  return (
+    <div>
+      <h1>Error</h1>
+      <p>{error.message}</p>
+      <pre>{error.stack}</pre>
+    </div>
+  )
+}
+
 /**
  * Main App component. Wrap the main react-router components together with
  * the various dialogs and error messages that may be needed.
  */
-const App = ({ history, config, appReady }) => {
-  if (!config.get('loginToken')) { // login is missing
+
+class App extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      hasError: false,
+    }
+  }
+
+  componentDidCatch (error, info) {
+    console.error(error, info)
+    this.setState({ hasError: true, error: error, errorInfo: info })
+  }
+
+  render () {
+    const { history, config, appReady } = this.props
+    const { hasError, error } = this.state
+
+    if (!config.get('loginToken')) { // login is missing
+      return (
+        <Grid fluid>
+          <NoLogin error={error} />
+        </Grid>
+      )
+    }
+
+    if (hasError) {
+      return (
+        <Grid fluid>
+          <Error />
```

Comment:
You need the `error` prop here.

What about the `errorInfo` from `componentDidCatch`?  That should probably be given to the error boundary report component as well.